language-design.md

```diff
@@ -74,7 +212,17 @@ implementation choices from users?
 
 **Response**
 
+Implementation choices change how user expressions are interpreted. Implementation choices can change how verbose or concise expressions in the language are. In this way, the feel of the language can be changed through the designer's choices. Good implementation choices give users intuitive results, a natural language feel, and an intuitive and sensible understanding of how the programming language realizes the user's ideas. We believe that good implementation choices will also lead to an interface that suggests what the implementation is to the user without the user needing to refer to documentation. (This is the notion that the language should read like documentation.) We do not believe that implementation choices should be hiden from the users. The documentation for a langauge should include implementation choices to enable users to make effective programing decisions. However, users should be able to comfortably program without detailed knowledege of the implementation. From our experiences, programming languages seem to imply their implementation details through their grammar, or syntax, so we do not believe that it is always possible to hide out implementation choices from users. Language designers should show the implementation choices so that users can use the language most effectively. Knowledgeable users can then provide feedback to the designer to help the language become the best that it can.
```

Comment:
I disagree, to some extent, with 

> programming languages seem to imply their implementation details through their grammar, or syntax

Suppose you have the following loop:

```
for item in collection:
    print item
```

I would argue that there are at least three straightforward ways to implement an interpreter or compiler that executes this code - recursion, looping, and actually figuring out every element in the collection and printing it a la

```
print collection__item1
print collection__item2
...
```

Just looking at this code doesn't tell you how the underlying implementation handles this, or even that it will always handle a for loop the same way.